tests/integration/targets/vmware_inventory_esxi_hosts/tasks/main.yml

```
@@ -0,0 +1,29 @@
+---
+- block:
+    - name: Import common vars
+      ansible.builtin.include_vars:
+        file: ../group_vars.yml
+
+    - name: Run Inventory Plugin
+      ansible.builtin.command: ansible-inventory -i "{{ role_path }}/files/test.esxi_hosts.yml" --list
+      register: _inventory_out
+
+    - name: Parse Inventory Results as JSON
+      ansible.builtin.set_fact:
+        inventory_results: "{{ _inventory_out.stdout | from_json }}"
+
+    # you can't reference the 'all' property here for some reason. It reverts back to the test playbook inventory
+    # instead of the inventory_results
+    - name: Check Output
+      ansible.builtin.assert:
+        that:
+          - first_host.ansible_host
+          - first_host.tags is defined
```

```suggestion
          - first_host.tags is defined and first_host.tags is not none
```
If tags is defined but has a None value, tags is defined will still pass.
To catch a None value explicitly, might be better to add and tags is not none ?

---

as well for ansible_host

---

thats a good idea. For some reason I wasnt able to test for None, but I am able to test if the variable is a dictionary. So I added that. I also added a check to make sure ansible_host is an IP

---

Great, thanks

---

I just noticed the regex (^[\d+\.]+$) only works for IPv4 and doesn’t handle IPv6. Should we assume first_host.ansible_host will always be IPv4, or do we need to update the regex to support IPv6 too?

---

we should check for ipv6 too!

---

ok :) so, just fix that and I think the PR is ok
thank you

---

`- first_host.ansible_host is regex('^((\\d{1,3}\\.){3}\\d{1,3}|([0-9a-fA-F:]+))$')`

smth like that?